arrow-docs/docs/io/README.md

```diff
@@ -12,3 +12,206 @@ Arrow has adopted suspend as system to model monadic computations and offers the
+Additionally, `IO` offers powerful concurrent operators and cancellation in addition of offering a referential transparent runtime.
+These properties are what makes using `IO` powerful, and `suspend` offers the exact same properties but natively in the language with support from the compiler. 
+
+Below we discuss a couple topics why we favor `suspend` over `IO`.
+
+### Ergonomics
+
+`IO` requires a wrapper in the return type: `fun number(): IO<Int>`, and thus we always have to work with the `IO` type to access the value we care about within.
+A typical pattern for this using `flatMap`, so let's say we want to calculate 3 numbers and return them as a `Triple`.
+
+```kotlin
+fun number(): IO<Int> = IO.just(1)
+
+fun triple(): IO<Triple<Int, Int, Int>> =
+  number().flatMap { a ->
+    number().flatMap { b ->
+      number().map { c ->
+        Triple(a, b, c)
+      }
+    }
+  }
+```
+
+So simply to call a function 3 times, and combine the result into a `Triple` we had to use `flatMap` twice and `map`.
+What that means under the hood we'll discuss in the performance section but in terms of ergonomics this is not ideal.
+Especially not if we can compare it to the following `suspend` code:
+
+```kotlin
+suspend fun number(): Int = 1
+
+suspend fun triple(): Triple<Int, Int, Int> = 
+  Triple(number(), number(), number())
+```
+
+Here we can see that we can simply forget about `flatMap` & `map` and we can simply construct the `Triple` and call `number()` three times directly in the constructor.
+
+The ergonomics of `suspend` are clearly better here, and this is a very important point in Kotlin since the language aims for high ergonomics and developer friendly constructs.
+
+### Safety / Purity / Referential transparency
+
+So what guarantees does `IO` bring, and does `suspend` offer the same guarantees?
+
+`IO<A>` always results in `Either<Throwable, A>` whether we run it with `unsafeRunSync` (throws Throwable), `unsafeRunAsync` or `unsafeRunAsyncCancellable`.
+This is done so that any exceptions that occur within the `IO` API can be safely returned to the user, and it can be recovered from at any point in the code.
+Important here is that a `Throwable` that occurs in an async thread is safely captured in the `IO` as well and can be recovered from at any point.
+
+`suspend` always results in `Result<A>` which is isomorphic to `Either<Throwable, A>`, and it can be used to offer the same safety guarantees as `IO`.
+So the `suspend` API can also always return any exception safely to the user, and it can be recovered from at any point in the code.
+In contrast to `IO`, we can only find `startCoroutine` in the standard library and it's isomorphic to `unsafeRunAsync`.
+Instead of `f: (Either<Throwable, A>) -> Unit` you provide `f: (Result<A>) -> Unit` to run the `suspend () -> A` program.
+
+So `suspend () -> A` offers us the exact same guarantees as `IO<A>`.
+
+Both also forces us to go through certain set of methods to execute/run the effectful program.
```

Comment:
Nit: 
can we mention which methods or a link to some docs? Or is this rather a narrative style to lead to the next section or topics which are discussed, if it's the latter, maybe we can make this a bit more clear.